### ISSUE TITLE
bug: Incorrect report of a misspelled word

### ISSUE THREAD

**Issue by Skoti** [pawamoy/mkdocs-spellcheck#27]
### Description of the bug

Words that are inside of a markdown formatting and use hyphens or apostrophes are reported as misspelled.

### To Reproduce

Markdown words like:
```
_global-level_
_field's name_
```

### Full traceback

```python
WARNING -  mkdocs_spellcheck: (symspellpy) mobile-ui/formatting/ios.md: Misspelled '_global', did you mean 'global'?
WARNING -  mkdocs_spellcheck: (symspellpy) mobile-ui/formatting/ios.md: Misspelled '_field', did you mean 'field', 'afield'?
```

### Expected behavior

Valid words that use hyphens or apostrophes inside markdown formatting like `_`, `__`, `*`, or `**` are not reported as misspelled.

### Environment information

```bash
python -m mkdocs_spellcheck.debug  # | xclip -selection clipboard
- __System__: macOS-15.3.2-arm64-arm-64bit
- __Python__: cpython 3.12.0
- __Environment variables__:
- __Installed packages__:
  - `mkdocs-spellcheck` v1.1.0
```
